src/lein_monolith/task/with_dependency_set.clj

```diff
@@ -0,0 +1,55 @@
+(ns lein-monolith.task.with-dependency-set
+  (:require
+    [clojure.java.io :as io]
+    [lein-monolith.plugin :as plugin]
+    [lein-monolith.task.util :as u]
+    [leiningen.core.main :as lein]
+    [leiningen.core.project :as project]))
+
+
+(defn- maybe-remove-dependencies
+  [project opts]
+  (if (:only opts)
+    (assoc project :dependencies '())
+    project))
+
+
+(defn- reload-project
+  "Reloads the raw project.clj file for the given project.
+   This has the effect of removing any middleware that was added to the project
+   as well as any changes to the project map that were made in the current
+   session."
+  [project]
+  (-> project
+      (:root)
+      (io/file "project.clj")
+      (str)
+      (project/read-raw)))
+
+
+(defn- change-dependencies
+  [project dependency-set]
+  (let [[monolith _] (u/load-monolith! project)
+        dependencies (plugin/get-dependencies-from-set! monolith dependency-set project)]
+    (if (:monolith project)
+      (assoc project :managed-dependencies dependencies)
+      (assoc project :monolith/dependency-set dependency-set))))
+
+
+(defn- setup
+  [project opts dependency-set]
+  (-> project
+      (reload-project)
+      (change-dependencies dependency-set)
+      (project/init-project)
+      (maybe-remove-dependencies opts)))
```

Comment:
Given the triviality of the logic in `maybe-remove-dependencies`, it seems like this line could be a `cond->` instead. The namespace is small enough that it's still reasonable to have separated, though. 🤔 No strong feeling either way here, but it's useful to consider the tradeoff between having to jump somewhere else to find out what a function does vs being able to read it directly inline.